### ISSUE TITLE
retnzestimate?

### ISSUE THREAD

**Issue by nitrocode** [seme0021/python-zillow#28]
Noticed this misspelling in a few places and it might be setting the wrong flag in `GetSearchResults()` and `GetZEstimate()` functions

https://github.com/seme0021/python-zillow/blob/master/zillow/api.py#L55

**Comment by bhoscheit** [comment]
Hi All,

I was wondering if anyone has had success using the rent zestimate via the API.

Simply trying: "data = api.GetSearchResults(key, address, postal_code, retnzestimate=True)" followed by "data.get_dict()" does not give me the rent zestimate for a particular property.

Thanks,
BH

**Comment by nitrocode** [comment]
@bhoscheit I don't think spamming all the issues is the best way to get help

**Comment by RyanConway91** [comment]
Any updates on this? I am trying to get rent estimates for properties which I know have them (can view them through the zillow app) but cannot seem to get this information from the API wrapper. Any idea how to fix this? I am assuming the "retnzestimate"typo is part of the issue, as with "retnzestimate" you would not be accessing the "rentzestimate" property of API. I fixed these typos in the code I downloaded but still not getting the results.

**Comment by bhoscheit** [comment]
Hey Ryan!

What project are you working on?

- Ben

**Comment by RyanConway91** [comment]
Hey Ben, got it working! Found the updated code and it is working now. What do you mean "project"? 

Thanks

**Comment by bhoscheit** [comment]
Great! I'm glad. I'm just curious/interested in what you are using the code
for.

- Ben

**Comment by RyanConway91** [comment]
Oh okay I thought maybe "project" that was github lingo for something. I am trying automate an realestate investment calculator for the Minneapolis area.

**Comment by bhoscheit** [comment]
Ah, very funny! I was doing the same for the Madison, WI area when I was
using this code a few months ago. Now the automatic ranking system netted
us our first investment property. Well, goodluck with either your future
product or investing! I would be interested in knowing how it pans out:
bhoscheit5@gmail.com

Thanks!
Ben

**Comment by RyanConway91** [comment]
Thanks!

**Comment by blackmarkt** [comment]
Hey @RyanConway91 If you don't mind me asking how did you get the `retnzestimate`/`rentestimate` to work? I have tried setting the parameter to `True` and I have the latest and only version. Thanks!

**Comment by RyanConway91** [comment]
api = zillow.ValuationApi()
        data = api.GetDeepSearchResults(key, address, postal_code,rentzestimate = True)
        data_dict = data.get_dict()
        bdrms = data_dict['extended_data']['bedrooms']
        bthrms = data_dict['extended_data']['bathrooms']
        sqft = data_dict['extended_data']['bathrooms']
        lot_sqft = data_dict['extended_data']['lot_size_sqft']
        tax_ass = float(data_dict['extended_data']['tax_assessment'])
        tax_rate = 1.81/100
        tax = tax_ass*tax_rate

        rent_zest = data_dict['rentzestimate']['amount']
        rent_zest_high = data_dict['rentzestimate']['valuation_range_high']
        rent_zest_low = data_dict['rentzestimate']['valuation_range_low']

I may have modified the package code I don’t remember. If this doesn’t work let me know


   Ryan J. Conway
   Hydrogeologist
   Minneapolis, MN office: 952.832.2846
   cell: 715.338.4386
   RConway@barr.com
   www.barr.com

[cid:image001.png@01D52D7D.A47412D0]<www.barr.com>

If you no longer wish to receive marketing e-mails from Barr, respond to communications@barr.com and we will be happy to honor your request.

**Comment by pmdbt** [comment]
@RyanConway91 Hey Ryan, thanks for putting this repo together. I followed your code here:

```python
api = zillow.ValuationApi()
        data = api.GetDeepSearchResults(key, address, postal_code,rentzestimate = True)
        data_dict = data.get_dict()
        bdrms = data_dict['extended_data']['bedrooms']
        bthrms = data_dict['extended_data']['bathrooms']
        sqft = data_dict['extended_data']['bathrooms']
        lot_sqft = data_dict['extended_data']['lot_size_sqft']
        tax_ass = float(data_dict['extended_data']['tax_assessment'])
        tax_rate = 1.81/100
        tax = tax_ass*tax_rate

        rent_zest = data_dict['rentzestimate']['amount']
        rent_zest_high = data_dict['rentzestimate']['valuation_range_high']
        rent_zest_low = data_dict['rentzestimate']['valuation_range_low']
```

However, I was getting an error:

```
TypeError: GetDeepSearchResults( ) got an unexpected keyword argument 'rentzestimate'
```

So, I looked at your source code and it appears you've made a syntax/spelling error on this line:

```python
def GetDeepSearchResults(self, zws_id, address, citystatezip, retnzestimate=False):
```

You spelled rentzestimate as retnzestimate. The method "GetSearchResult()" has the same typo as well. Thought you should know and make a quick update :). I'm using version 0.2.0 for reference.

@nitrocode pointed this out in his original comment as well, but it doesn't appear to be fixed everywhere.

**Comment by jecronin** [comment]
@RyanConway91 circling back on @nitrocode and @pmdbt's notes. How is the spelling issue affecting the ability to pull in rentzestimate from the API? For example, if you go to this URL the rentzestimate is displayed in the XML but it is not when you call the Valuation API in python. The rent zestimate with the price would be great to do ROI calcs at scale for a batch of properties, all within python Thank you for your help!

https://www.zillow.com/webservice/GetSearchResults.htm?zws-id=X1-ZWz1h9tnvngv7v_31kch&address=11820+Melones+Circle&citystatezip=Gold+River%2c+CA&rentzestimate=True


![image](https://user-images.githubusercontent.com/45111583/85905708-f9286780-b7c0-11ea-9cf2-2c72629eab0c.png)

**Comment by nitrocode** [comment]
It might be time to fork this repo or use another zillow package or another realestate api package

**Comment by pmdbt** [comment]
Agreed, I'm actually using my own version where I just changed all of the spelling errors and it works fine. If others are interested in volunteering and maintaining a fork, I'm happy to do it and make a commit to fix the spelling issues.

Sent via Superhuman ( https://sprh.mn/?vip=jerrychu123@gmail.com )

**Comment by RyanConway91** [comment]
Yes, you need to fix the typos for the rentzestimate to work


   Ryan J. Conway
   Hydrogeologist
   Minneapolis, MN office: 952.832.2846
   cell: 715.338.4386
   RConway@barr.com
   www.barr.com

[cid:image001.png@01D64C6F.FB02BDB0]<www.barr.com>

If you no longer wish to receive marketing e-mails from Barr, respond to communications@barr.com and we will be happy to honor your request.

**Comment by jecronin** [comment]
@pmdbt that would be great thank you! Sidebar: I talked to Max awhile back and he took me through Lofty. Amazing product, awesome work

**Comment by jecronin** [comment]
@RyanConway91, I've submitted a pull request to update the retnzestimate typos to rentzestimate. Thank you for your help!

**Comment by thaskef** [comment]
It would be amazing if we can get the rent estimate fixed.  As is, everything works in a Google Colab notebook except for that.  I wish someone could help so I don't have to go back and redo it all on a local machine.
